src/cli/commands/joinNetwork.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import prompts from 'prompts';
+
+import { ArweaveSigner, IOToken, mIOToken } from '../../node/index.js';
+import { JoinNetworkOptions } from '../types.js';
+import {
+  formatIOWithCommas,
+  jwkToAddress,
+  requiredJwkFromOptions,
+  writeIOFromOptions,
+} from '../utils.js';
+
+export async function joinNetwork(options: JoinNetworkOptions) {
+  const jwk = requiredJwkFromOptions(options);
+  const address = jwkToAddress(jwk);
+  const io = writeIOFromOptions(options, new ArweaveSigner(jwk));
+
+  const {
+    disableDelegatedStaking,
+    disableAutoStake,
+    delegateRewardShareRatio,
+    fqdn,
+    label,
+    minDelegatedStake,
+    note,
+    observerAddress,
+    port,
+    properties,
+    operatorStake,
+    allowedDelegates,
+  } = options;
+
+  if (label === undefined) {
+    throw new Error(
+      'Label is required. Please provide a --label for your node.',
+    );
+  }
+  if (fqdn === undefined) {
+    throw new Error('FQDN is required. Please provide a --fqdn for your node.');
+  }
+  if (operatorStake === undefined) {
+    throw new Error(
+      'Operator stake is required. Please provide a --operator-stake denominated in IO for your node.',
+    );
+  }
+
+  const ioQuantity = new IOToken(+operatorStake);
+  const mIOOperatorStake = ioQuantity.toMIO().valueOf();
+
+  const settings = {
+    observerAddress,
+    operatorStake: mIOOperatorStake,
+    allowDelegatedStaking:
+      disableDelegatedStaking === undefined
+        ? undefined
+        : !disableDelegatedStaking,
+    autoStake: disableAutoStake === undefined ? undefined : !disableAutoStake,
+    delegateRewardShareRatio:
+      delegateRewardShareRatio !== undefined
+        ? +delegateRewardShareRatio
+        : undefined,
+    allowedDelegates,
+    fqdn,
+    label,
+    minDelegatedStake:
+      minDelegatedStake !== undefined ? +minDelegatedStake : undefined,
+    note,
+    port: port !== undefined ? +port : undefined,
+    properties,
+  };
+
+  if (!options.skipConfirmation) {
+    const balance = await io.getBalance({ address });
```

Comment:
write commands take a --skip-confirmation boolean. by default we'll do some user friendly assertions and ask the user to approve the action before writing to the AO network